### PR TITLE
chore:  enhancements in User Service and User common service

### LINKED ISSUE #4278: Enhancement: Optimisations and Refactoring around User Service.
### 🔖 Feature description

1. We are hitting multiple db calls while deleting the roles from casbin when removing permission during delete app flow, instead it should be done in bulk.
2. Refactoring around query for getRolebyFilterForAllTypes - using query builder of gopg.
3. Introducing comments in different flows.
4. Introducing type constants to add one layer of protection.
5. Improving split logic to avoid removing placeholder in loops .
6. Avoid using strings.toLower at multiple places.
7. Doing bulk db operations where required.
8. Separate around function when build role filters for user and rolegroups.

### 🎤 Pitch

Refactoring should be done to optimise and make the code more scalable and readable.

### ✌️ Solution

Refactoring around user service and permissions.

### 🔄️ Alternative

_No response_

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/devtron-labs/devtron/blob/main/CODE_OF_CONDUCT.md)

AB#6959

### PR DESCRIPTION
**Pull request by Shivam-nagar23**
# Description

Fixes #4278 

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code.
* [x] I have commented my code, particularly in hard-to-understand areas.
* [x] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: internal/sql/repository/app/AppRepository.go [hunk 1]
```diff
@@ -452,7 +452,7 @@ func (repo AppRepositoryImpl) FetchAppIdsByDisplaynames(names []string) (map[int
 		DisplayName string `json:"display_name"`
 	}
 	var jobIdName []App
-	whereCondition := " where active = true and app_type = 2 "
+	whereCondition := fmt.Sprintf(" where active = true and app_type = %v ", helper.Job)
```
Comment (kartik-579) [inline internal/sql/repository/app/AppRepository.go:455]: if this method is only used for jobs, lets rename it

File: pkg/user/UserAuthService.go [hunk 2]
```diff
@@ -496,45 +496,56 @@ func (impl UserAuthServiceImpl) DeleteRoles(entityType string, entityName string
...
+			}
+			for _, rl := range allUsersMappedToRoles {
+				success := casbin2.DeleteRoleForUser(rl, roleModel.Role)
+				if !success {
+					impl.logger.Warnw("error in deleting casbin policy for role", "role", roleModel.Role)
+					casbinDeleteFailed = append(casbinDeleteFailed, success)
+				}
+			}
+		}
+		// removing all policies for the role
```
Comment (kartik-579) [inline pkg/user/UserAuthService.go:523]: move into new method

File: pkg/user/UserCommonService.go [hunk 3]
```diff
@@ -629,3 +580,95 @@ func (impl UserCommonServiceImpl) GetCapacityForRoleFilter(roleFilters []bean.Ro
...
+	case bean.CLUSTER_ENTITIY:
+		{
+			impl.BuildRoleFilterKeyForCluster(roleFilterMap, role, key)
+		}
+	case bean2.EntityJobs:
+		{
+			impl.BuildRoleFilterKeyForJobs(roleFilterMap, role, key)
+		}
+	default:
+		{
```
Comment (kartik-579) [inline pkg/user/UserCommonService.go:595]: redundant braces i guess, please remove if so

File: pkg/user/repository/bean.go [hunk 4]
```diff
@@ -24,3 +24,7 @@ const (
 	WorkflowPValUpdateKey     PValUpdateKey = "Workflow"
 	WorkflowObjPValUpdateKey  PValUpdateKey = "WorkflowObj"
 )
+
+const (
+	EMPTY = ""
```
Comment (kartik-579) [inline pkg/user/repository/bean.go:29]: empty what. empty is subjective term....values can be same but definition of empty can be different for different usecases

Comment (kartik-579) [reply pkg/user/repository/bean.go:29]: also where is this used, can not find usages

**Review by kartik-579 — CHANGES_REQUESTED** [review]

File: internal/sql/repository/app/AppRepository.go (hunk 1 above)
Comment (Shivam-nagar23) [reply internal/sql/repository/app/AppRepository.go:455]: DONE

File: pkg/user/UserAuthService.go (hunk 2 above)
Comment (Shivam-nagar23) [reply pkg/user/UserAuthService.go]: DONE

File: pkg/user/repository/bean.go (hunk 4 above)
Comment (Shivam-nagar23) [reply pkg/user/repository/bean.go]: Renamed it accordingly. Used as empty placeholder in orm query arguments at multiple places

**Review by nishant-d — APPROVED** [review]

**Review by kartik-579 — APPROVED** [review]

**Pull request merged**